hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/RexToMongoVisitor.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.mongodb;
+
+import com.hazelcast.sql.impl.expression.Expression;
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.rex.RexCorrelVariable;
+import org.apache.calcite.rex.RexDynamicParam;
+import org.apache.calcite.rex.RexFieldAccess;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexLiteral;
+import org.apache.calcite.rex.RexLocalRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexOver;
+import org.apache.calcite.rex.RexPatternFieldRef;
+import org.apache.calcite.rex.RexRangeRef;
+import org.apache.calcite.rex.RexSubQuery;
+import org.apache.calcite.rex.RexTableInputRef;
+import org.apache.calcite.rex.RexVisitor;
+import org.apache.calcite.util.NlsString;
+import org.bson.conversions.Bson;
+
+import java.util.List;
+
+/**
+ * Visitor that converts REX nodes to Mongo expressions.
+ */
+final class RexToMongoVisitor implements RexVisitor<Object> {
+
+    private static final Object[] EMPTY_EXPRESSION_OPERANDS = new Object[0];
+
+    private final String[] fields;
+
+    RexToMongoVisitor(String[] fields) {
+        this.fields = fields;
+    }
+
+    @Override
+    public Object visitInputRef(RexInputRef inputRef) {
+        int index = inputRef.getIndex();
+        return fields[index];
```

Comment:
If have a call that has 2 operands, first is inputRef - in RexToExpressionVisitor this inputRef will be a ColumnExpression, here it will be a string and call will get a string, which we will use as a column name in MongoDB's filter. 

I can wrap it around some temporary type "ColumnRef" and later in visitCall I can unwrap it, if that will be clearer